wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsSQLClient.java

```diff
@@ -344,21 +344,28 @@ public HPCCTableWrapper[] getTables(@SpanAttribute String filter) throws Excepti
             log.error("HPCCWsSQL.getTables(" + filter + ") encountered RemoteException.\n" + e.getLocalizedMessage());
         }
 
-        if (resp.getExceptions() != null)
-            handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could not get Tables(" + filter + ").");
-
-        if (resp.getTables() != null)
+        if (resp != null)
         {
-            HPCCTable[] hpccTables = resp.getTables().getTable();
-            if (hpccTables != null && hpccTables.length > 0)
+            if (resp.getExceptions() != null)
+                handleEspExceptions(new ArrayOfEspExceptionWrapper(resp.getExceptions()), "Could not get Tables(" + filter + ").");
+
+            if (resp.getTables() != null)
             {
-                result = new HPCCTableWrapper[hpccTables.length];
-                for (int i = 0; i < hpccTables.length; i++)
+                HPCCTable[] hpccTables = resp.getTables().getTable();
+                if (hpccTables != null && hpccTables.length > 0)
                 {
-                    result[i] = new HPCCTableWrapper(hpccTables[i]);
+                    result = new HPCCTableWrapper[hpccTables.length];
+                    for (int i = 0; i < hpccTables.length; i++)
+                    {
+                        result[i] = new HPCCTableWrapper(hpccTables[i]);
+                    }
                 }
             }
         }
+        else
```

Comment:
Is it valid to have no tables in the response?

Reply:
> Is it valid to have no tables in the response?

0 tables is valid, null response isn't and is logged and null returned.